lib-ee/emqx_license/src/emqx_license_parser_v20220101.erl

```diff
@@ -103,35 +109,49 @@ parse_payload(Payload) ->
     end.
 
 parse_type(TypeStr) ->
-    case string:to_integer(TypeStr) of
-        {Type, <<"">>} -> {ok, Type};
+    case parse_int(TypeStr) of
+        {ok, Type} -> {ok, Type};
         _ -> {error, invalid_license_type}
     end.
 
 parse_customer_type(CTypeStr) ->
-    case string:to_integer(CTypeStr) of
-        {CType, <<"">>} -> {ok, CType};
+    case parse_int(CTypeStr) of
+        {ok, CType} -> {ok, CType};
         _ -> {error, invalid_customer_type}
     end.
 
-parse_date_start(<<Y:4/binary, M:2/binary, D:2/binary>>) ->
-    Date = list_to_tuple([N || {N, <<>>} <- [string:to_integer(S) || S <- [Y, M, D]]]),
-    case calendar:valid_date(Date) of
-        true -> {ok, Date};
-        false -> {error, invalid_date}
+parse_date_start(DateStr) ->
+    case parse_int(DateStr) of
+        {ok, Num} when Num >= ?MIN_START_DATE ->
+            Y = Num div 1_00_00,
+            M = (Num rem 1_00_00) div 1_00,
+            D = Num rem 1_00,
```

Comment:
Nit: maybe group by 3 digits?

```suggestion
            Y = Num div 10_000,
            M = (Num rem 10_000) div 100,
            D = Num rem 100,
```

Reply:
intended to group by 2 digits because it's 2 digits for month and 2 digits for day.